src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableActivityContext.cs

```diff
@@ -1,145 +1,145 @@
-                    Activator.CreateInstance(destinationType) :
-                    null;
-            }
-
-            JToken jToken = this.GetInputAsJson();
-            var value = jToken as JValue;
-            if (value != null)
-            {
-                return value.ToObject(destinationType);
-            }
-
-            string serializedValue = jToken.ToString(Formatting.None);
-
-            // Object inputs for out-of-proc activities are passed in their JSON-stringified form with a destination
-            // type of System.String. Unfortunately, deserializing a JSON string to a string causes
-            // MessagePayloadDataConverter to throw an exception. This is a workaround for that case. All other
-            // inputs with destination System.String (in-proc: JSON and not JSON; out-of-proc: not-JSON) inputs with
-            // destination System.String should cast to JValues and be handled above.)
-            if (destinationType.Equals(typeof(string)))
-            {
-                return serializedValue;
-            }
-
-            return MessagePayloadDataConverter.Default.Deserialize(serializedValue, destinationType);
-        }
-
-        internal string GetSerializedOutput()
-        {
-            return this.serializedOutput;
-        }
-
-        /// <summary>
-        /// Sets the JSON-serializeable output of the activity function.
-        /// </summary>
-        /// <remarks>
-        /// If this method is not called explicitly, the return value of the activity function is used as the output.
-        /// </remarks>
-        /// <param name="output">
-        /// The JSON-serializeable value to use as the activity function output.
-        /// </param>
-        internal void SetOutput(object output)
-        {
-            if (this.serializedOutput != null)
-            {
-                throw new InvalidOperationException("The output has already been set of this activity instance.");
-            }
-
-            if (output != null)
-            {
-                JToken json = output as JToken;
-                if (json != null)
-                {
-                    this.serializedOutput = json.ToString(Formatting.None);
-                }
-                else
-                {
-                    this.serializedOutput = MessagePayloadDataConverter.Default.Serialize(output);
-                }
-            }
-            else
-            {
-                this.serializedOutput = null;
-            }
-        }
-    }
-}
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// The default parameter type for activity functions.
+    /// </summary>
+    internal class DurableActivityContext : IDurableActivityContext
```

Comment:
How does moving from `DurableActivityContextBase` to `IDurableActivityContext` affect unit testing, ex. the example unit tests in our samples?

Reply:
It breaks everything, I am afraid.

I have already updated all the unit tests in the solution, though of course there may be a lot of other places that I am not aware of.